### PR TITLE
HLSL-Preprocessor fixes

### PR DESCRIPTION
**Pull request by AnastaZIuk**

### CODE REVIEW

File: src/nbl/asset/utils/CHLSLCompiler.cpp [hunk 1]
```diff
@@ -328,8 +329,19 @@ std::string CHLSLCompiler::preprocessShader(std::string&& code, IShader::E_SHADE
...
             stream << i->get_value();
+            
+            // TODO: replace this hack with `support_option_emit_contnewlines` flag when Boost::Wave respect it
+            // See also https://github.com/Devsh-Graphics-Programming/Nabla/issues/746
+            if (emit_nl_after != -1) {
+                emit_nl_after--;
+                if (emit_nl_after == 0) stream << "\n";
+            }
+            if (boost::ends_with(stream.str(), "#pragma shader_stage(")) emit_nl_after = 2;
+        }
```
Comment (devshgraphicsprogramming) [inline src/nbl/asset/utils/CHLSLCompiler.cpp:344]: don't we use Boost Wave in GLSL compiler as well?

Comment (alichraghi) [reply src/nbl/asset/utils/CHLSLCompiler.cpp:344]: no we don't

Comment (devshgraphicsprogramming) [reply src/nbl/asset/utils/CHLSLCompiler.cpp]: yeah I just foundout with @nipunG314 today XD

**Comment by devshgraphicsprogramming** [comment]
@Erfan-Ahmadi merge if it pleases you.

**Pull request merged**